### PR TITLE
Addition of CUB finance

### PR DESCRIPTION
**Pull request by notakian**
##### Twitter Link:
https://twitter.com/financeleo

##### List of audit links if any:
https://www.certik.org/projects/cubfinance

##### Website Link:
https://cubdefi.com/

##### Logo (High resolution, preferably in .svg and .png, for application on both white and black backgrounds. Will be shown with rounded borders):
https://cubdefi.com/images/cub/token.svg

##### Current TVL:
$4m

##### Chain:
BSC

##### Coingecko ID (so your TVL can appear on Coingecko): (https://api.coingecko.com/api/v3/coins/list)
cub-finance
{"id":"cub-finance","symbol":"cub","name":"Cub Finance"}
##### Coinmarketcap ID (so your TVL can appear on Coinmarketcap): (https://api.coinmarketcap.com/data-api/v3/map/all?listing_status=active,inactive,untracked&start=1&limit=10000)
8858
{"id":8858,"name":"Cub Finance","symbol":"CUB"}

##### Short Description (to be shown on DefiLlama):
Leofinance, the parent of Cubfinance, is an Established Project with a 2 Year Track Record of Building Blockchain-Based Applications. DeFi Yield Farming apps are a dime a dozen these days. Cub Finance is different because it is built as a branch of our LeoFi organization underneath the LeoFinance project.

The goal of Cub Finance is to adopt and build DeFi-based smart contracts for the entire LeoFinance project. Through BEP20 LEO (bLEO), CUB and the adoption of existing use cases in the space, we're using the basic starting point of the Cub Finance yield platform to build a new generation of DeFi apps to expand and service our community of thousands of MAUs.

LeoFinance started as a platform for crypto and finance content creators. Our flagship application - LeoFinance.io - connects the community on a web3 social media platform. Think of it as being similar to the popular Web2 blogging application "Medium" but on the blockchain.

##### Token address and ticker if any:
0x50d809c74e0b8e49e7b4c65bb3109abe3ff4c1c1

##### Category (Yield/Dexes/Lending/Minting/Assets/Insurance/Options/Indexes/Staking) *Please choose only one:
Yield

##### forkedFrom (Does your project originate from another project):
Goose Finance and Autofarm

##### methodology (what is being counted as tvl, how is tvl being calculated):
Cub Finance has farms and what are call kingdoms on the platform (which are auto-compounding farms of other platforms). 
All farms are included, except those that pool the CUB token. 
All kingdoms are included except one, which is the CUB token kingdom.

Where the farm is an LP token, that is decomposed into its underlying assets, where it is a single asset farm, that asset is being used.

### CODE REVIEW

File: projects/cubfinance/index.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	// farms
+    const poolLength = (await sdk.api.abi.call({
+        target: cubFarmAddress,
+        abi: abi['poolLength'],
+        chain:'bsc',
+      })).output;
+
+      const lps = []
+    for(var i = first_pool ; i < poolLength ; i++){
+        if (!farmsExcluded.includes(i)){
```
Comment (0xngmi) [inline projects/cubfinance/index.js:32]: would it be better to exclude based on symbol?

**Comment by 0xngmi** [comment]
Looks good, is it just 2 masterchefs with native staking & pool2 removed?

**Comment by notakian** [comment]
Yes, that's it for now. I considered excluding based on Token, but then half of the LP pool would be included (the BUSD and BNB components).

**Comment by notakian** [comment]
also, just checking did you want me to change the code to exclude based on symbol, or are you comfortable the way it is?

**Comment by 0xngmi** [comment]
no need :)

**Pull request merged**